jdaviz/app.py

```diff
@@ -1162,18 +1162,21 @@ def simplify_spectral_subset(self, subset_name, att, overwrite=False):
             Whether to update the current subset with the simplified state or apply it
             to a new subset.
         """
-        spectral_region = self.get_subsets(subset_name, spectral_only=True)
-        new_state = None
-        # Reverse through sub regions so that they are added back
-        # in the order of lowest values to highest
-        for index in range(len(spectral_region) - 1, -1, -1):
-            convert_to_range = RangeSubsetState(spectral_region[index].lower.value,
-                                                spectral_region[index].upper.value,
-                                                att)
-            if new_state is None:
-                new_state = convert_to_range
-            else:
-                new_state = new_state | convert_to_range
+        if self.is_there_overlap_spectral_subset(subset_name):
+            new_state = self.merge_overlapping_spectral_regions(subset_name, att)
```

Comment:
does `merge_overlapping_spectral_regions` also account for other simplification?  In other words, would the user have to call/press this twice if they have a composite subset with both overlapping and "removed chunks"?

Reply:
Good catch, working on this now.

Reply:
Tested and this does in fact work.